scalafmt-core/shared/src/main/scala/org/scalafmt/config/SortSettings.scala

```diff
@@ -1,24 +1,34 @@
 package org.scalafmt.config
 
 import metaconfig._
+import scala.meta.classifiers.Classifier
+import scala.meta.Tree
+import scala.meta.Mod
+import sourcecode.Name
+import sourcecode.Text
 
 case class SortSettings(
     order: List[SortSettings.ModKey]
 )
 
 object SortSettings {
-
-  implicit val SortSettingsModKeyReader: ConfCodec[ModKey] =
-    ReaderUtil.oneOf[ModKey](
-      `implicit`,
-      `final`,
-      `sealed`,
-      `abstract`,
-      `override`,
-      `private`,
-      `protected`,
-      `lazy`
-    )
+  final case class ModKey(
```

Comment:
I based this PR on top of #2285 because they have conflicting changes in `build.sbt` and `ScalafmtRunner.scala`. I will try to separate Scala 3 PRs in the future. I suspect it will be easier to separate PRs once once the `Scala3` dialect changes in this PR are merged.